### PR TITLE
Feature buy album

### PR DESCRIPTION
**Pull request by christiangelone**
## Summary
* Covers that an logged user/admin can buy an album
* Covers that a logged user/admin cannot buy the same album twice
* Covers that an user/admin cannot buy a not found album

## Known Issues
 * No issues

## Trello Card
 * https://trello.com/c/EN1B6dZw

### CODE REVIEW

File: app/controllers/albums.js [hunk 1]
```diff
@@ -13,3 +14,26 @@ exports.all = (req, res, next) => {
       next(errors.externalApiError(err));
     });
 };
+
+exports.buy = (req, res, next) => {
+  const albumId = req.params.id;
+  return AlbumsService.getById(albumId)
+    .then(album => {
+      AlbumPurchase.make(req.user.id, album.id)
```
Comment (gonziesc) [inline app/controllers/albums.js]: return

File: app/models/albumpurchase.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    'AlbumPurchase',
+    {
+      id: { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
+      userId: { type: DataTypes.INTEGER, unique: 'purchase_unique' },
+      albumId: { type: DataTypes.INTEGER, unique: 'purchase_unique' }
+    },
+    {}
+  );
+
+  AlbumPurchase.make = (userId, albumId) => {
```
Comment (gonziesc) [inline app/models/albumpurchase.js]: why make and not create?

File: app/models/albumpurchase.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      albumId
+    }).catch(err => {
+      if (err.name === 'SequelizeUniqueConstraintError') {
+        return Promise.reject(new Error('You cannot buy the same album twice'));
+      }
+      return Promise.reject(err.name);
+    });
+  };
+
+  return AlbumPurchase;
```
Comment (gonziesc) [inline app/models/albumpurchase.js]: use camelCase

**Review by gonziesc — CHANGES_REQUESTED** [review]

File: app/models/albumpurchase.js (hunk 2 above)
Comment (christiangelone) [reply app/models/albumpurchase.js]: Because it already exists (Already defined in sequelize) and enters in conflict

Comment (gonziesc) [reply app/models/albumpurchase.js]: just use the sequelize create and manage errors in another part

File: app/models/albumpurchase.js (hunk 3 above)
Comment (christiangelone) [reply app/models/albumpurchase.js]: It was written that way by sequelize model generator ¯\_(ツ)_/¯

File: app/models/albumpurchase.js (hunk 2 above)
Comment (gonziesc) [reply app/models/albumpurchase.js]: you are not doing anything else

File: app/controllers/albums.js (hunk 1 above)
Comment (christiangelone) [reply app/controllers/albums.js]: done!

File: app/models/albumpurchase.js (hunk 3 above)
Comment (gonziesc) [reply app/models/albumpurchase.js]: ok

File: app/models/albumpurchase.js (hunk 2 above)
Comment (christiangelone) [reply app/models/albumpurchase.js]: Done, error handling moved to controller

**Review by gonziesc — APPROVED** [review]

**Pull request merged**